### PR TITLE
feat(CTFtime.org): add new presence

### PR DESCRIPTION
**Pull request by ELJoOker2004**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/C/CTFtime/metadata.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		"id": "605894319408283678",
+		"name": "eljooker"
+	},
+	"service": "CTFtime",
+	"description": {
+		"en": "CTFtime is a website that tracks cybersecurity competitions known as Capture The Flag (CTF) events. These contests challenge participants to solve a variety of tasks, ranging from scavenger hunts on Wikipedia to basic programming exercises and even hacking into servers to steal data. "
+	},
+	"url": "ctftime.org",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/ZHkLk9I.png",
```
Comment (DarkVillager) [inline websites/C/CTFtime/metadata.json:13]: Logo should be 512x512

File: websites/C/CTFtime/presence.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
+type Functionlize<T> = {
+	[P in keyof T]: () => T[P];
+};
+
+interface Route extends Functionlize<Partial<PresenceData>> {
+	path: RegExp;
+}
+
+const enum Assets {
+	Logo = "https://i.imgur.com/ZHkLk9I.png",
```
Comment (DarkVillager) [inline websites/C/CTFtime/presence.ts:10]: Logo should be 512x512

File: websites/C/CTFtime/presence.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+			},
+			{
+				path: /^\/event.*$/,
+				details: () => "Viewing a CTF",
+				state: () => document.querySelector(".page-header h2").textContent,
+				largeImageKey: () => {
+					const imgElements = document.querySelectorAll("span2");
+					return `https://ctftime.org/${imgElements[0]
+						.querySelector("img")
+						.getAttribute("src")}`;
```
Comment (DarkVillager) [inline websites/C/CTFtime/presence.ts:131]:
```suggestion
imageElement ?? Assets.Logo
```

File: websites/C/CTFtime/presence.ts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+enum Settings {
+	TIMESTAMP = "timestamp",
+	BUTTONS = "buttons",
+	COVERS = "images",
+}
+const { href, hostname } = document.location,
+	presence = new Presence({
+		clientId: "1253040451624370300",
+	}),
```
Comment (DarkVillager) [inline websites/C/CTFtime/presence.ts:21]:
```suggestion
	}),
	imageElement = `https://ctftime.org/${document.querySelector<htmlImageElement>("span2").querySelector("img").src}`;
```

File: websites/C/CTFtime/presence.ts [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+			{
+				path: /^\/user.*$/,
+				details: () => "Viewing user profile",
+				state: () => document.querySelector(".page-header h2").textContent,
+				largeImageKey: () => {
+					const imgElements = document.querySelectorAll("span2");
+					return `https://ctftime.org/${imgElements[0]
+						.querySelector("img")
+						.getAttribute("src")}`;
+				},
```
Comment (DarkVillager) [inline websites/C/CTFtime/presence.ts:151]:
```suggestion
					imageElement ?? Assets.Logo
				},
```

File: websites/C/CTFtime/presence.ts [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+			{
+				path: /^\/team.*$/,
+				state: () => document.querySelector(".page-header h2").textContent,
+				details: () => "Viewing a Team",
+				largeImageKey: () => {
+					const imgElements = document.querySelectorAll("span2");
+					return `https://ctftime.org/${imgElements[0]
+						.querySelector("img")
+						.getAttribute("src")}`;
+				},
```
Comment (DarkVillager) [inline websites/C/CTFtime/presence.ts:170]:
```suggestion
imageElement ?? Assets.Logo
				},
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/C/CTFtime/presence.ts (hunk 4 above)
Comment (ELJoOker2004) [reply websites/C/CTFtime/presence.ts:21]: i don't see a point of doing this + it doesn't work

Comment (ELJoOker2004) [reply websites/C/CTFtime/presence.ts:21]: nvm done

File: websites/C/CTFtime/metadata.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.10",
+	"author": {
+		"id": "605894319408283678",
+		"name": "eljooker"
+	},
+	"service": "CTFtime",
+	"description": {
+		"en": "CTFtime is a website that tracks cybersecurity competitions known as Capture The Flag (CTF) events. These contests challenge participants to solve a variety of tasks, ranging from scavenger hunts on Wikipedia to basic programming exercises and even hacking into servers to steal data. "
```
Comment (Bas950) [inline websites/C/CTFtime/metadata.json:9]:
```suggestion
		"en": "CTFtime is a website that tracks cybersecurity competitions known as Capture The Flag (CTF) events. These contests challenge participants to solve a variety of tasks, ranging from scavenger hunts on Wikipedia to basic programming exercises and even hacking into servers to steal data."
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
For the rest LGTM

**Review by DarkVillager — APPROVED** [review]

**Review by Timeraa — APPROVED** [review]

**Pull request merged**